moveit_ros/planning/plan_execution/src/plan_execution.cpp

```diff
@@ -357,7 +357,7 @@ moveit_msgs::msg::MoveItErrorCodes plan_execution::PlanExecution::executeAndMoni
     // in the meantime we do a hack:
 
     bool unwound = false;
-    for (std::size_t j = 0; j < i; ++j)
+    for (int j = i - 1; j >= 0; --j)
```

Comment:
Let's use longer names to improve readability

```suggestion
    for (size_t prev_component = current_component - 1; prev_component >= 0; --prev_component)
```

Reply:
Overall, I'm not sure out this approach. Won't each component be "unwound" repeatedly? Once at the for-loop of L352, then again at L360...

Reply:
@mechwiz what would you think about splitting these `executeAndMonitor` changes to a different PR, so they don't hold up getting the other bug fix merged?

Reply:
I don't think so since there is a "break" at L369

Reply:
I see. Basically the for-loop at L360 goes backward until it finds an earlier component for the same group, then unwinds it. So it can skip components which are not for the same group. Is that right?

Reply:
correct - just to clarify to make sure we are on the same page "it finds an earlier component for the same group, then unwinds it" -> "it finds an earlier component for the same group, then unwinds the current component based on the last waypoint of the earlier component". yep it should skip any  components not of the same group.

Reply:
I'm OK with making a separate PR but feel that the time difference between getting each of these issues resolved won't be significant to make it worth it.

Reply:
Thanks for the explanation, I'm OK with it now. Shame on MoveIt for not having some basic tests for `unwind()`

Reply:
on a side note - I could be wrong but I think the compiler might complain about using `size_t` here since the value of `j`/`prev_component` on the first pass is - 1.